qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -100,37 +116,120 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
+									codeActionResolveFutures,
+									resolver,
+									sharedSettings);
+							break;
+						default:
+							break;
 					}
 				}
 			}
 		}
-		return CompletableFuture.completedFuture(codeActions);
+		CompletableFuture<Void>[] registrationsArray = new CompletableFuture[codeActionResolveFutures.size()];
+		codeActionResolveFutures.toArray(registrationsArray);
+		return CompletableFuture.allOf(registrationsArray).thenApply((Void _void) -> {
+			return codeActions;
+		});
+	}
+
+	private void doCodeActionToCreateProperty(Template template, Diagnostic diagnostic, QuteErrorCode errorCode,
+			List<CodeAction> codeActions, List<CompletableFuture<Void>> registrations,
+			QuteTemplateGenerateMissingJavaMember resolver, SharedSettings settings) {
+
+		UnknownPropertyData unknownPropertyData = DiagnosticDataFactory.getMissingMemberData(diagnostic);
+		if (unknownPropertyData == null) {
+			return;
+		}
+		String missingProperty = unknownPropertyData.getProperty();
+		String resolvedType = unknownPropertyData.getSignature();
+		String propertyCapitalized = missingProperty.substring(0, 1).toUpperCase() + missingProperty.substring(1);
+
+		String projectUri = template.getProjectUri();
+
+		GenerateMissingJavaMemberParams publicFieldParams = new GenerateMissingJavaMemberParams(MemberType.Field,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams getterParams = new GenerateMissingJavaMemberParams(MemberType.Getter,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams appendToTemplateExtensionsParams = new GenerateMissingJavaMemberParams(
+				MemberType.AppendTemplateExtension, missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams createTemplateExtensionsParams = new GenerateMissingJavaMemberParams(
+				MemberType.CreateTemplateExtension, missingProperty, resolvedType, projectUri);
+		
+
+		CodeAction createPublicField = createCodeActionWithData(
+				String.format(CREATE_PUBLIC_FIELD, missingProperty, resolvedType), publicFieldParams,
+				Collections.singletonList(diagnostic));
+		CodeAction createGetter = createCodeActionWithData(
+				String.format(CREATE_GETTER, propertyCapitalized, resolvedType), getterParams,
+				Collections.singletonList(diagnostic));
+		CodeAction appendToTemplateExtensions = createCodeActionWithData(String
+				.format(APPEND_TO_TEMPLATE_EXTENSIONS, missingProperty),
+				appendToTemplateExtensionsParams, Collections.singletonList(diagnostic));
+		CodeAction createTemplateExtensions = createCodeActionWithData(String.format(CREATE_TEMPLATE_EXTENSIONS, missingProperty), createTemplateExtensionsParams, Collections.singletonList(diagnostic));
+		codeActions.add(createPublicField);
+		codeActions.add(createGetter);
+		codeActions.add(appendToTemplateExtensions);
+		codeActions.add(createTemplateExtensions);
+
+		registrations.add(resolver.resolveTemplateCodeAction(publicFieldParams) //

```

Comment:
You can do that in an another PR, but it should be really nice to support resolve code actions. Please create an issue for that.

Reply:
https://github.com/redhat-developer/quarkus-ls/issues/604